### PR TITLE
Gossip config: Get the Gossip Memberlist config from the pilosa config file.

### PR DESCRIPTION
**Pull request by travisturner**
## Overview

Get the Gossip Memberlist config from the pilosa config file. 
Added: PushPull interval, Gossip Interval, Probe Interval config options

### CODE REVIEW

**Comment by raskle** [comment]
@travisturner This PR does have the default port setting commit, so the feature will work on a single node.  My question is do we want to document these settings, or force users to use our defaults since they controls the fundamental gossip operation.

**Comment by jaffee** [comment]
is this PR still valid, or superseded by later cluster config work @travisturner ?

**Comment by travisturner** [comment]
The concept is still valid, but the `config.go` structure has changed since this PR was created (`Gossip` is a now a sub struct within the `Config` struct), so it's going to require some modification.

**Comment by travisturner** [comment]
The memberlist configuration options that differ between the LAN and WAN configs (in `memberlist/config.go`) are:

`LAN`:
```
    conf.TCPTimeout = 10 * time.Second
    conf.SuspicionMult = 4
    conf.PushPullInterval = 30 * time.Second
    conf.ProbeTimeout = 500 * time.Millisecond
    conf.ProbeInterval = 1 * time.Second
    conf.GossipNodes = 3
    conf.GossipInterval = 200 * time.Millisecond
    conf.GossipToTheDeadTime = 30 * time.Second
```

`WAN`:
```
    conf.TCPTimeout = 30 * time.Second
    conf.SuspicionMult = 6
    conf.PushPullInterval = 60 * time.Second
    conf.ProbeTimeout = 3 * time.Second
    conf.ProbeInterval = 5 * time.Second
    conf.GossipNodes = 4 // Gossip less frequently, but to an additional node
    conf.GossipInterval = 500 * time.Millisecond
    conf.GossipToTheDeadTime = 60 * time.Second
```
so we should look at each of those and consider adding config support for them.

File: config.go [hunk 1]
```diff
@@ -41,17 +41,29 @@ const (
...
 	// DefaultMaxWritesPerRequest is the default number of writes per request.
 	DefaultMaxWritesPerRequest = 5000
+
+	// DefaultGossipPushPullInterval is the push pull interval.
+	DefaultGossipPushPullInterval = 15 * time.Second
+
+	// DefaultGossipInterval is the Gossip interval.
+	DefaultGossipInterval = 100 * time.Millisecond
+
+	// DefaultGossipProbeInterval is the Gossip Probe intercal.
```
Comment (travisturner) [inline config.go]: intercal > interval

**Comment by travisturner** [comment]
I incorporated this work in to #1014

**Pull request closed without merging**